app/controllers/user_applications_controller.rb

```diff
@@ -171,13 +180,41 @@ def get_application_volunteer_choices
     @third_choice_volunteer_position = VolunteerPosition.find(@user_application.third_choice_volunteer_position_id) unless @user_application.third_choice_volunteer_position_id.nil?
   end
 
-  def change_status(params, status, message)
-    @user_application = UserApplication.find(params['id'])
-    authorize @user_application
+  def change_status(status, message)
     @user_application.user_application_status_id = UserApplicationStatus.find_by(status: status).id
```

Comment:
@SeanLF you might look into what happens if the UserApplicationStatus is not found. You might get a null pointer by calling "id" on it.